src/newrelic/app_test.go

```diff
@@ -238,13 +238,19 @@ func TestNeedsConnectAttempt(t *testing.T) {
 		t.Fatal(now, app.lastConnectAttempt, app.state)
 	}
 
-	app.state = AppStateInvalidLicense
+	app.state = AppStateDisconnected
 	app.lastConnectAttempt = now.Add(-limits.AppConnectAttemptBackoff)
 	if app.NeedsConnectAttempt(now, limits.AppConnectAttemptBackoff) {
 		t.Fatal(now, app.lastConnectAttempt, app.state)
 	}
 
-	app.state = AppStateDisconnected
+	app.state = AppStateRestart
+	app.lastConnectAttempt = now.Add(-limits.AppConnectAttemptBackoff)
+	if app.NeedsConnectAttempt(now, limits.AppConnectAttemptBackoff) {
+		t.Fatal(now, app.lastConnectAttempt, app.state)
+	}
+
+	app.state = AppStateInvalidLicense
```

Comment:
This test seems to be testing the backoff for retries if an invalid license response is received from the collector - are we doing retries in this situation?